### PR TITLE
Extend quantifier API to support user specified weights

### PR DESCRIPTION
**Pull request by maurobringolf**
Currently quantifiers are always built with the default weight 0, this PR extends the quantifier constructors with an argument for the weight.

### CODE REVIEW

File: src/Z3/Base.hs [hunk 1]
```diff
@@ -1866,6 +1868,7 @@ marshalMkQ z3_mk_Q ctx pats x s body = marshal z3_mk_Q ctx $ \f ->
 -- variable with index 0, the second to last element of /xs/ refers to the
 -- variable with index 1, etc.
 mkForall :: Context
+          -> Int        -- ^ quantifiers are associated with weights indicating the importance of using the quantifier during instantiation. By default, pass the weight 0.
```
Comment (IagoAbal) [inline src/Z3/Base.hs:1871]: Prefer to not introduce breaking changes unless it's really necessary. I think we should rename `mkForall` (and so on) as `mkForallW` and then make `mkForall` a simple call to `mkForallW` with the weight argument set to `0`.

**Review by IagoAbal — CHANGES_REQUESTED** [review]
Thanks for another PR!

File: src/Z3/Base.hs (hunk 1 above)
Comment (maurobringolf) [reply src/Z3/Base.hs:1871]: I thought about that too, lets do it then :+1:

**Review by IagoAbal — APPROVED** [review]
Thanks!

**Pull request merged**